crates/gateway-client/src/lib.rs

```diff
@@ -103,6 +103,118 @@ pub trait GatewayApi: Sync {
     }
 }
 
+#[async_trait::async_trait]
+impl<T: GatewayApi + Sync + Send> GatewayApi for std::sync::Arc<T> {
```

Comment:
This looks very wrong from a design perspective.

This is also unnecessary (wrapping `gateway_client::Client` in `Arc`) as it is easily `Clone`able easily. The instance of `reqwest` client itself has `Arc` under the hood.

Reply:
Yes, but this is to enable `Clone` for tests using mockall. I agree its ugly and dumb and I would rather get rid of mockall entirely. There is no other way from what I can tell.

Reply:
`gateway::Client` isn't `Arc`'d only the mockall instances used in the tests are.

I'm open to alternatives - that don't include significantly rewriting things. That can happen, but in a separate PR.

Reply:
Got it, the whole `Arc` thing is for mocking.

I'd stay away from mocks as long as I can, but this is a topic for a whole different conversation.